src/ape/api/providers.py

```diff
@@ -80,12 +103,54 @@ def convert_parent_hash(cls, data):
         data["parentHash"] = parent_hash
         return data
 
+    @model_validator(mode="wrap")
+    @classmethod
+    def validate_size(cls, values, handler):
+        """
+        A validator for handling non-computed size.
+        Saves it to a private member on this class and
+        gets returned in computed field "size".
+        """
+
+        if not hasattr(values, "pop"):
+            # Handle weird AttributeDict missing pop method.
+            # https://github.com/ethereum/web3.py/issues/3326
+            values = {**values}
+
+        size = values.pop("size", None)
+        model = handler(values)
+        if size is not None:
+            model._size = size
+
+        return model
+
     @computed_field()  # type: ignore[misc]
     @cached_property
     def transactions(self) -> List[TransactionAPI]:
+        """
+        All transactions in a block.
+        """
         query = BlockTransactionQuery(columns=["*"], block_id=self.hash)
         return cast(List[TransactionAPI], list(self.query_manager.query(query)))
 
+    @computed_field()  # type: ignore[misc]
+    @cached_property
+    def size(self) -> int:
+        """
+        The size of the block in gas. Most of the time,
+        this field is passed to the model at validation time,
+        but occassionally it is missing (like in Infura::newHeads),
+        in which case it gets calculated.
```

Comment:
```suggestion
        in which case it gets calculated if and only if the user
        requests it (or during serialization of this model to disk).
```

Reply:
This _is_ at the abstract API layer. Technically, who's to say how downstream providers will handle this. Yes, ape-ethereum does this but it is kind of an assumption otherwise.